bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -1641,6 +1742,20 @@ private static void execBinaryOpCodes(WorkerExecutionContext ctx, WorkerData sf,
                 k = operands[2];
                 sf.doubleRegs[k] = sf.doubleRegs[i] / sf.doubleRegs[j];
                 break;
+            case InstructionCodes.DDIV:
+                i = operands[0];
+                j = operands[1];
+                k = operands[2];
+                operand1 = ((BDecimal) sf.refRegs[i]).decimalValue();
+                operand2 = ((BDecimal) sf.refRegs[j]).decimalValue();
+                if (operand2.compareTo(BigDecimal.ZERO) == 0) {
```

Comment:
shall we remove this and add isNaN support for decimal support? may be in a different PR?

Reply:
Created a new issue to track this (https://github.com/ballerina-platform/ballerina-lang/issues/11385). Since BigDecimal does not have NaN or Infinity, supporting isNaN, isFinite, isInfinite builtin functions in Ballerina won't be straightforward.